### ISSUE TITLE
Doesn't work for me in many cases

### ISSUE THREAD

**Issue by Drugoy** [einaregilsson/Redirector#105]
Case 1:
https://gif.d3.ru/ - on this site there are a lot of posts with embedded videos from i.imgur.com (just ctrl+f 'i.imgur.com' to find them).
I have a rule to force https for imgur requests:
```
        {
            "description": "[http>https] imgur",
            "exampleUrl": "http://i.imgur.com/IjrS5hL.png",
            "exampleResult": "https://i.imgur.com/IjrS5hL.png",
            "error": null,
            "includePattern": "http://*.imgur.com/*",
            "excludePattern": "",
            "redirectUrl": "https://$1.imgur.com/$2",
            "patternType": "W",
            "processMatches": "noProcessing",
            "disabled": false,
            "appliesTo": [
                "main_frame",
                "sub_frame",
                "stylesheet",
                "script",
                "image",
                "object",
                "xmlhttprequest",
                "other"
            ]
        }
```
But when I try to play any imgur video on that site - I get this entry in the errors console:
Loading mixed (insecure) display content “http://i.imgur.com/p9lmt1q.mp4” on a secure page

So, obviously, the requests are being sent to http.

Case 2:
Come without cookies to https://market.yandex.ru/ and open developer's console on 'network' tab and look for requests, then make a few clicks on the page and monitor the requests: there'll be a POST request to `https://market.yandex.ru/api/data-collector?*`.
This rule should redirect that request to localhost:
```
        {
            "description": "Cure Ya.Market (anti-spy)",
            "exampleUrl": "https://market.yandex.ru/api/data-collector?sk=yee4a92c5cfbfea2009d278c5e4643314",
            "exampleResult": "http://127.0.0.1/",
            "error": null,
            "includePattern": "https://market.yandex.ru/api/data-collector?*",
            "excludePattern": "",
            "redirectUrl": "http://127.0.0.1/",
            "patternType": "W",
            "processMatches": "noProcessing",
            "disabled": false,
            "appliesTo": [
                "main_frame",
                "sub_frame",
                "stylesheet",
                "script",
                "image",
                "object",
                "xmlhttprequest",
                "other"
            ]
        }
```
But it fails 😒

**Comment by Drugoy** [comment]
I found the solution: https://addons.mozilla.org/firefox/addon/urlredirector/

**Comment by Drugoy** [comment]
Now I've found an even better solution: https://addons.mozilla.org/firefox/addon/requestcontrol/
https://github.com/tumpio/requestcontrol

**Comment by einaregilsson** [comment]
Well, I'm glad you found something that works for you :)  Not really working on Redirector any more, so no idea if these problems will ever be fixed.

**Comment by ohohohoo** [comment]
I tried hotlinking a random image using http from a simple html file and it gets redirected to https.

![Scrshot](https://i.imgur.com/LtFpo3r.jpg)

**Comment by ZenFi** [comment]
I put "imgur.com" to Smart HTTPS and now *.imgur.com/* always loads https.

**Comment by Drugoy** [comment]
From what I've gathered - Redirector fails to work with POST requests.

After trying other add-ons a bit longer - I figured out that each has its flaws :(

**Comment by alexolog** [comment]
@Drugoy , can you summarize your findings?

**Comment by Drugoy** [comment]
@alexolog TL;DR is that currently there's no ultimate solution for redirecting requests based on regex rules matching request urls and query payload (in case of POST requests). From my point of view, the best what there currently is - is tumpio's requestcontrol. It has complex and weird interface.

**Comment by gkrishnaks** [comment]
@Drugoy , I saw your mention about POST requests.

Redirector specifically redirects GET requests only - to avoid accidentally redirecting POST requests.

See this highlight : https://github.com/einaregilsson/Redirector/blob/master/js/background.js#L49-L53

If you are sure about letting through POST requests, maybe you can clone the repo and edit this highlighted block (perhaps with some additional conditions like details.url matching your URL  so that you don't universally redirect POST requests but only for your particular case) and build the addon for your personal use.

**Comment by ZIEXED** [comment]
We need an option "Handle Post requests" per rule or global

**Comment by Drugoy** [comment]
1st step is to just treat POSTs as GETs and redirect them the same way
2nd step add per-option control for POSTs. I would switch browser for such an add-on.

**Comment by gkrishnaks** [comment]
@ZIEXED Global rule to enable it or disable - while having it disabled by default upon install - I can send a PR for that... I am not sure how we can do it for "per-rule" ?

**Comment by ZIEXED** [comment]
@gkrishnaks That would be great for now :)
about the per rule option if you figured it out later please do it
@einaregilsson I know u don't have time to work on this project anymore, maybe you can just tell us your opinion about this? (per-rule)

**Comment by einaregilsson** [comment]
Hi. I don't want to deal with POST redirects. I'm not even sure the webRequest API redirects them with the form data included, it might just do a redirect which causes a GET request to be made. In any case, I prefer to only handle GET requests, I don't want the additional complexity of POST request handling.
